x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -108,11 +108,9 @@ public AggregatorImplementer(Elements elements, TypeElement declarationType, Int
             (declarationType.getSimpleName() + "AggregatorFunction").replace("AggregatorAggregator", "Aggregator")
         );
         this.valuesIsBytesRef = BYTES_REF.equals(TypeName.get(combine.getParameters().get(combine.getParameters().size() - 1).asType()));
-        intermediateState = Arrays.stream(interStateAnno).map(state -> new IntermediateStateDesc(state.name(), state.type())).toList();
+        intermediateState = Arrays.stream(interStateAnno).map(IntermediateStateDesc::newIntermediateStateDesc).toList();
```

Comment:
I merged the `IntermediateStateDesc` from `AggregatorImplementer` and `GroupingAggregatorImplementer` because they share a lot in common already and share more after this change.

Sadly this does touch *every* grouping aggregator to make the commit noisier. I'm going to review this change before un-drafting the PR.